Client/Application/UserConversionMetrics.swift

```diff
@@ -0,0 +1,97 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import Foundation
+import Shared
+import StoreKit
+
+class UserConversionMetrics {
+    private static var appOpenTimestamps: [Date] {
+        get { return UserDefaults.standard.array(forKey: "appOpenTimestamps") as? [Date] ?? [Date]() }
+        set { UserDefaults.standard.set(newValue, forKey: "appOpenTimestamps") }
+    }
+
+    private static var searchesTimestamps: [Date] {
+        get { return UserDefaults.standard.array(forKey: "searchesTimestamps") as? [Date] ?? [Date]() }
+        set { UserDefaults.standard.set(newValue, forKey: "searchesTimestamps") }
+    }
+
+    public static func didStartNewSession() {
+        if UserConversionMetrics.shouldRecordMetric() {
+            UserConversionMetrics.appOpenTimestamps.append(Date())
+            UserConversionMetrics.checkProfileActivation()
+        }
+    }
+
+    public static func didPerformSearch() {
+        if UserConversionMetrics.shouldRecordMetric() {
+            UserConversionMetrics.searchesTimestamps.append(Date())
+            UserConversionMetrics.checkProfileActivation()
+        }
+    }
+
+    // there shouldn't be more records added to UserDefaults once this event happens and it should not affect existing users
+    private static func shouldRecordMetric() -> Bool {
+        guard let firstAppUse = UserDefaults.standard.object(forKey: PrefsKeys.Session.FirstAppUse) as? UInt64 else {
+            return false
+        }
+        if UserDefaults.standard.bool(forKey: "didUpdateConversionValue") {
+            return false
+        }
+        let firstAppUseDate = Date.fromTimestamp(firstAppUse)
+        let oneWeekSinceFirstUse = Calendar.current.date(byAdding: .day, value: 7, to: firstAppUseDate)!
```

Comment:
This should be an if let, we should avoid using force unwrap

Reply:
The force unwrap is still there it should be fixed like it was in `shouldActivateProfile` func